File: Semanticus.Engine/DaxErrorText.cs
using System.Text.RegularExpressions;

namespace Semanticus.Engine
{
    /// <summary>Strips Analysis Services object-markup tags (olii, ccon, and similar) from a query error so both
    /// doors show the names a person can read, not the raw tags.</summary>
    public static class DaxErrorText
    {
        static readonly Regex Markup = new Regex(@"</?[A-Za-z]{2,8}>", RegexOptions.Compiled);

        public static string Plain(string text)
        {
            if (string.IsNullOrEmpty(text)) return text;
            return Markup.Replace(text, "");
        }
    }
}

File: Semanticus.VSCode/webview/src/daxErrorText.ts
/** Drop Analysis Services markup tags such as olii so a query error shows the object name, not the tags. */
export function plainDaxError(text: string | null | undefined): string {
    if (!text) return text ?? '';
    return text.replace(/<\/?[A-Za-z]{2,8}>/g, '');
}
